Vwitter/UserCell.m

```diff
@@ -0,0 +1,102 @@
+//
+//  UserCell.m
+//  Vwitter
+//
+//  Created by Christina Li on 7/11/22.
+//
+
+#import "UserCell.h"
+#import "Follow.h"
+
+@implementation UserCell
+
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    // Initialization code
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+
+    // Configure the view for the selected state
+}
+
+- (void)setUser:(PFUser *)user{
+    _user = user;
+
+    [self refreshData];
+
+}
+
+- (void)refreshData {
+
+    NSString *at = @"@";
+    self.username.text = [NSString stringWithFormat:@"%@%@", at, self.user.username];
+    
+    if ([self.user.objectId isEqualToString:[PFUser currentUser].objectId]) {
+        self.followStatusButton.hidden = YES;
+    }
+    
+    [PFCloud callFunctionInBackground:@"existsFollow"
+                       withParameters:@{@"currentUserObjectId":[PFUser currentUser].objectId, @"followingUserObjectId":self.user.objectId}
+                                block:^(id exists, NSError *error) {
+      if (!error) {
+          NSLog (@"%d", ((NSNumber *)exists).boolValue);
+          if (((NSNumber *)exists).boolValue) {
+              [self.followStatusButton setTitle:@"Following" forState:UIControlStateNormal];
+          }
+          else {
+              [self.followStatusButton setTitle:@"Follow" forState:UIControlStateNormal];
+          }
+      }
+    }];
+}
+
+- (IBAction)didFollow:(id)sender {
+
+    PFQuery *thisFollow = [Follow query];
+    [thisFollow whereKey:@"followingUserObjectId" equalTo:self.user.objectId];
+    [thisFollow whereKey:@"currentUserObjectId" equalTo:[PFUser currentUser].objectId];
+    
+    __weak typeof(self) weakSelf = self;
+    [thisFollow findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
+        typeof(self) strongSelf = weakSelf;
+        if (!strongSelf) {
+            NSLog(@"I got killed!");
+            return;
+        }
```

Comment:
great job doing the weak/strong self dance! 

**Problem 1**: simple views should never have business logic in them, and this is even more important for UITableView cells, because they get "reused" for performance reasons. Next week, let's discuss how you can implement the "delegate" pattern. This will allow you to keep this "UserCell" view class as a simple view class, who only worries about UI, and instead, whenever the `followButton` is pressed, this user cell will call something like `[self.delegate didPressFollowForUser:self.user]`, and in your view controller, you will implement this method and do the actual network request. 

The delegate pattern will allow this UserCell class to call a method on its "owner" without having to actually know what class the owner is, because in the future, there might be several different view controller classes in your app that could be using this UserCell to show users. 

**Problem 2**: the nested network requests. This should be a simple Parse Cloud function, something like `toggleFollowStatus` which accepts a `userId` as a parameter. This will make the client (iOS) logic significantly simpler